src/rules/media-query-value-dollar-variable/index.js

```diff
@@ -0,0 +1,62 @@
+import { utils } from "stylelint"
+import { namespace } from "../../utils"
+
+export const ruleName = namespace("media-query-value-dollar-variable")
+
+export const messages = utils.ruleMessages(ruleName, {
+  rejected: "Unexpected dollar-variable as a value in a media query",
+  expected: "Expected a dollar-variable (e.g. $var) to be used as a value in a media query",
+})
+
+export default function (expectation, options) {
+  return function (root, result) {
+    const validOptions = utils.validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [ "always", "never" ],
+    }, {
+      actual: options,
+      optional: true,
+      possible: {
+        ignore: [ "calculations", "funcitons" ],
+      },
+    })
+    if (!validOptions) { return }
+
+    // In `(max-witdh: 10px )` find `: 10px )`
+    const valueRegex = /:(?:\s*?)(\S.+?)(:?\s*?)\)/
+    const valueRegexGlobal = new RegExp(valueRegex.source, "g")
```

Comment:
Is there any reason to use the regex constructor here? You could just put the `g` option to the regex literal.